### PR TITLE
components

### PR DESCRIPTION
**Pull request by carlos-ssh**
- Add background at App.js.
- Add operate and calculate files that contain the logic for the calculator.
- Add big from npm.
- Add fontawesome icons.
- Add press button effect.

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    } catch (err) {
+      return '0';
+    }
+  }
+  if (operation === '%') {
+    try {
+      return one.mod(two).toString();
+    } catch (err) {
+      return '0';
+    }
```
Comment (D4nielJ) [inline src/logic/operate.js:39]: [OPTIONAL] I think it could be best if you customize the message when an error occurs in the calculator. That way users can understand what happens when they try to do a forbidden operation.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -29,78 +74,135 @@ class Calculator extends Component {
...
+                onClick={() => this.buttonPress('.')}
+              >
                 .
               </Button>
-              <Button className="button symbols">
-                =
+              <Button
+                className="button symbols"
+                onClick={() => this.buttonPress('=')}
+              >
```
Comment (D4nielJ) [inline src/components/Calculator.js]: - [ ] I think you can move all of this code to a new component ("CalculatorButtons") to avoid all of the unrelated code to the functionality of the calc. Also, you will practice how to compose components and how to pass props down.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @carlos-ssh, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: :fire: 
- Your calculator works as expected

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (carlos-ssh) [reply src/components/Calculator.js]: this requirement isn't in the project requirements where i can found more information related to it? @D4nielJ

Comment (carlos-ssh) [reply src/components/Calculator.js]: @D4nielJ

Comment (carlos-ssh) [reply src/components/Calculator.js]: @D4nielJ

Comment (D4nielJ) [reply src/components/Calculator.js]: [You can read more about it here](https://reactjs.org/docs/components-and-props.html).

This is from the project requirements:
> Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, lift state up.

Comment (carlos-ssh) [reply src/components/Calculator.js]: **@D4nielJ  I'm not using it?** 
> Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, lift state up.

**Review by dasileker — APPROVED** [review]
## **Hi @carlos-ssh,** 

### **Nice Work**
## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (D4nielJ) [reply src/components/Calculator.js]: Well, you are not passing props down since you built all of the components on the same file. I think knowing how to lift state and pass props down is a valuable skill when developing on react and also the point of this lesson.

Also, I reviewed the changes you made and I don't agree with them. You moved the entire component one level down, instead, you could try just moving the buttons from the calculator, and leaving the logic on the Calculator component.

Comment (carlos-ssh) [reply src/components/Calculator.js]: try to explain better which changes you want! :)

Comment (D4nielJ) [reply src/components/Calculator.js]: I'm sorry my explanation was not good enough. I will definitely take this into account for future reviews. Happy coding :sweat_smile:
